marconi-chain-index/src/Marconi/ChainIndex/Indexers/AddressDatum.hs

```diff
@@ -245,8 +249,9 @@ instance Buffered AddressDatumHandle where
     :: Foldable f
     => f (StorableEvent AddressDatumHandle)
     -> AddressDatumHandle
-    -> IO AddressDatumHandle
-  persistToStorage es h = do
+    -> StorableMonad AddressDatumHandle AddressDatumHandle
```

Comment:
What if the wrapping to `EitherT e IO` happened within `Marconi.Core.Storable`, such that each and every indexer wouldn't need to change its `StorableMonad` to list it out explicitly?

Reply:
It's the firt thing that I tried, but it implies roping SQLite in `core`, and was a nightmare to implement, I didn't keep track of why though.

Reply:
Ok, I gave it another shot and I remember why I did not do it: it's doable, but you don't do any error management in Storable.
And yes we don't change StorableMonad but we have to modify all the signature that has a StorableMonad in it and that are related to the storable module. So to save one change, we impose many to the type signatures.